pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -29,8 +31,387 @@ import (
+
+	// RequestedCapacity for the pool creation
+	RequestedCapacity resource.Quantity `json:"requestedCapacity,omitempty"`
+}
+
+// StoragePoolStatus is for handling status of StoragePool resource
+type StoragePoolStatus struct {
+	// ReferenceResource
+	ReferenceResource StoragePoolReferenceResource `json:"referenceResource,omitempty"`
+
+	// StorageCapacity of the pool,viz total, used and available capacity
+	StorageCapacity StoragePoolStorageCapacity `json:"storageCapacity,omitempty"`
+
+	// StorageIOPs of the pool,viz total, provisioned, used and available capacity
+	StorageIOPs StoragePoolStorageIOPs `json:"storageIops,omitempty"`
+
+	// DeviceConfiguration for list of Device UIDs and raid configurations currently active
+	DeviceConfiguration StoragePoolDeviceConfiguration `json:"deviceConfiguration,omitempty"`
+
+	//VolumeSizeMaxLimit for maximum volume size allowed
+	VolumeSizeMaxLimit resource.Quantity `json:"volumeSizeMaxLimit,omitempty"`
+
+	// State of the pool based on various scenarios
+	State StoragePoolState `json:"state,omitempty"`
+}
+
+// StoragePoolType of the storage pool, viz lvm, spdk blobstor
+type StoragePoolType string
+
+const (
+	// LVM2StoragePool for lvm pools
+	LVM2StoragePool StoragePoolType = "lvm2"
+
+	// SpdkBlobstorStoragePool for spdk-blobstor pools
+	SpdkBlobstorStoragePool StoragePoolType = "spdk-blobstor"
+)
+
+// StoragePoolCofiguration for the defined type of the pool
+type StoragePoolCofiguration struct {
+	// LVM pool related configuration
+	// +nullable
+	LVM *LVMPoolConfiguration
+
+	// SpdkBlobStor pool related configuration
+	// +nullable
+	SpdkBlobStor *SpdkBlobStorPoolConfiguration
+}
+
+// LVMPoolConfiguration for LVM pool related configuration
+type LVMPoolConfiguration struct {
+	//TODO: Decide the sub fields
+}
+
+// SpdkBlobStorPoolConfiguration for SpdkBlobStor pool related configuration
+type SpdkBlobStorPoolConfiguration struct {
+	//TODO: Decide the sub fields
+}
+
+// StoragePoolDeviceSpec for the configuration and filtering related specs for devices
+type StoragePoolDeviceSpec struct {
+	// MaxDeviceCount for maximum allowed devices
+	MaxDeviceCount uint64 `json:"maxDeviceCount,omitempty"`
+
+	// DeviceTypeIdentifier for the device viz.  ssd-4k, ssd-16k, nvme-64k
+	DeviceTypeIdentifier StoragePoolDeviceTypeIdentifier `json:"deviceTypeIdentifier,omitempty"`
+
+	// DeviceSelector for labels to match with devices incase they are labeled
+	DeviceSelector map[string]string `json:"deviceSelector,omitempty"`
+
+	// DeviceSelector for filtering devices based on FiltersOptions
+	DeviceFilters map[string][]FilterOptions `json:"deviceFilters,omitempty"`
+
+	// Devices for the UIDs of the free devices
+	Devices []types.UID `json:"devices,omitempty"`
+}
+
+type StoragePoolCapabilities struct {
+	// DataStorageCapabilities for the storage pool
+	DataStorageCapabilities StoragePoolDataStorageCapabilities `json:"dataStorageCapabilities,omitempty"`
```

Comment:
repetitive Capabilities can we avoid it?

Reply:
Sure. Removing Capabilities from Variable names